product-service/services/products.service.ts

```diff
@@ -0,0 +1,20 @@
+import * as products from '../data/products.json';
+
+export class ProductsService {
+  constructor() {}
+
+  public getAllProducts() {
+    return products;
+  }
+
+  public getProductById(id) {
+    return new Promise((resolve, reject) => {
+      const productById = products.filter(product => { return product.id === id })[0];
```

Comment:
I suppose, method 'find' will be more suitable.

Reply:
Thanks for suggestion, I'll rework this.